integration-test/src/main/java/com/sequenceiq/it/cloudbreak/action/ums/AssignResourceRoleUserAction.java

```diff
@@ -27,22 +27,21 @@ public AssignResourceRoleUserAction(String userKey) {
     @Override
     protected UmsTestDto umsAction(TestContext testContext, UmsTestDto testDto, UmsClient client) throws Exception {
         CloudbreakUser user = testContext.getRealUmsUserByKey(userKey);
+        String userCrn = user.getCrn();
         String resourceRole = testDto.getRequest().getRoleCrn();
         String resourceCrn = testDto.getRequest().getResourceCrn();
-        Log.when(LOGGER, format(" Assigning resource role '%s' at resource '%s' for user '%s' ", resourceRole, resourceCrn, user.getCrn()));
-        Log.whenJson(LOGGER, format(" Assign resource role request:%n "), testDto.getRequest());
 
-        Multimap<String, String> assignedResourceRoles = client.getDefaultClient().listAssignedResourceRoles(user.getCrn(), Optional.of(""));
-        LOGGER.info(format(" Assigned resource roles ['%s'] are present to user '%s' ", assignedResourceRoles, user.getCrn()));
+        Log.when(LOGGER, format(" Assigning resource role '%s' to user '%s' at resource '%s'... ", resourceRole, userCrn, resourceCrn));
+        Log.whenJson(LOGGER, format(" Assign resource role request:%n "), testDto.getRequest());
+        LOGGER.info(format(" Assigning resource role '%s' to user '%s' at resource '%s'... ", resourceRole, userCrn, resourceCrn));
+        Multimap<String, String> assignedResourceRoles = client.getDefaultClient().listAssignedResourceRoles(userCrn, Optional.of(""));
         if (assignedResourceRoles.get(resourceCrn).contains(resourceRole)) {
-            LOGGER.info(format(" Resource role '%s' has already been assigned to user '%s' for resource '%s' ", resourceRole, user.getCrn(), resourceCrn));
-            Log.when(LOGGER, format(" Resource role '%s' has already been assigned to user '%s' for resource '%s' ", resourceRole, user.getCrn(), resourceCrn));
+            LOGGER.info(format(" Resource role '%s' has already been assigned to user '%s' at resource '%s' ", resourceRole, userCrn, resourceCrn));
+            Log.when(LOGGER, format(" Resource role '%s' has already been assigned to user '%s' at resource '%s' ", resourceRole, userCrn, resourceCrn));
         } else {
-            client.getDefaultClient().assignResourceRole(user.getCrn(), resourceCrn, resourceRole, Optional.of(""));
-            // wait for UmsRightsCache to expire
-            Thread.sleep(7000);
-            LOGGER.info(format(" Resource role '%s' has been assigned at resource '%s' for user '%s' ", resourceRole, resourceCrn, user.getCrn()));
-            Log.when(LOGGER, format(" Resource role '%s' has been assigned at resource '%s' for user '%s' ", resourceRole, resourceCrn, user.getCrn()));
+            client.getDefaultClient().assignResourceRole(userCrn, resourceCrn, resourceRole, Optional.of(""));
+            LOGGER.info(format(" Resource role '%s' has been assigned to user '%s' at resource '%s' ", resourceRole, userCrn, resourceCrn));
```

Comment:
I'm fine with removing the Thread.sleep from here, but now we should somehow enforce that validation, because now it would be easy to misuse the framework since the assign and verification is not atomic.

```
                .when(umsTestClient.unAssignResourceRole(L0UserKeys.ENV_CREATOR_A))
                .then((tc, dto, client) -> validateAssignedResourceRole(tc, dto, client, userEnvCreatorA, getIamGroupAdminCrn(), false))
```

Reply:
My bad, these lines were removed mistakenly. So I've reintroduced the removed lines. Thanks for pointer!